### PR TITLE
.github: manifest.yml: Add CI labels based on modules in west.yml

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Add CI labels only when relevant modules are changed.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

Needs this PR in action-manifest: https://github.com/zephyrproject-rtos/action-manifest/pull/2
I'll update the commit sha in the `uses: zephyrproject-rtos/action-manifest@` line after that is merged

The label list is incomplete, need input on this.

The labelling can be tested by opening a PR to this branch, where you change west.yml.

### CODE REVIEW

File: .github/workflows/manifest.yml
Comment (mbolivar-nordic) [inline .github/workflows/manifest.yml:32]: Wht's up with the `>-` syntax; is that intentional?

Comment (oyvindronningstad) [reply .github/workflows/manifest.yml:32]: It denotes a multiline string.

Comment (mbolivar-nordic) [reply .github/workflows/manifest.yml:32]: Did you run into errors without the chomping indicator? I'm not aware of us using it in many places; I guess the underlying tool should be able to handle a trailing newline, right?

Comment (oyvindronningstad) [reply .github/workflows/manifest.yml:32]: It works fine in my tests. `>` strips all newlines except the trailing. `>-` strips all newlines including the trailing.

https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines

Comment (mbolivar-nordic) [reply .github/workflows/manifest.yml:32]: What I'm asking is if the chomping indicator (which is the YAML jargon for the `-` character in this case; see the YAML spec for details) is required. If not, I think it's weird to include it here and therefore bad for readability.

Comment (oyvindronningstad) [reply .github/workflows/manifest.yml:32]: Sorry, I misunderstood the jargon.
You're right, a newline should be fine. strip() defaults to removing whitespace according to the docs. I added the `-` since a newline is not called for here, without knowing any convention. I'll remove it.

    labels = [x.strip() for x in args.labels.split(',')] \
        if args.labels != 'none' else None

**Review by mbolivar-nordic — APPROVED** [review]
I didn't look at the action code, but I think this approach is very reasonable and data-driven.

File: .github/workflows/manifest.yml
Comment (mbolivar-nordic) [reply .github/workflows/manifest.yml]: Thanks!

I'm probably just hypersensitive to YAML multiline string differences because I have to periodically clean up the 'wrong' ones in the DT bindings and suggest changes in review (https://github.com/zephyrproject-rtos/zephyr/pull/36189; https://github.com/nrfconnect/sdk-nrf/pull/3576; https://github.com/zephyrproject-rtos/zephyr/pull/35357#discussion_r650289075; https://github.com/zephyrproject-rtos/zephyr/pull/35841#discussion_r648505680; ...).

There are so many ways to do it and it's nice to pick just a few if we can.

File: .github/workflows/manifest.yml
Comment (tejlmand) [inline .github/workflows/manifest.yml:34]: Could you add some more comments here on how to use.
I can deduce by myself, but I think it's nice to have a bit more description for other to read.

Also maybe update the commit message with info on which relationship this commit defines.

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: .github/workflows/manifest.yml
Comment (oyvindronningstad) [reply .github/workflows/manifest.yml:39]: Thanks @tejlmand. I added a comment with examples and instructions. I'm not sure what you are asking for the commit message, but I added a bit of rationale.

**Comment by oyvindronningstad** [comment]
@shantha-14 Who can help make the label list complete?

**Review by tejlmand — APPROVED** [review]
lgtm

**Comment by mbolivar-nordic** [comment]
> @shantha-14 Who can help make the label list complete?

I don't think we need to block the merge on this, since we can always adjust if we discover issues. Let's just get @shantha-14 to ACK the approach.

The remaining blocker is getting the action changes merged upstream and updating this PR not to point at your fork of the action repository, right?

**Review by thst-nordic — APPROVED** [review]

**Comment by oyvindronningstad** [comment]
>I don't think we need to block the merge on this, since we can always adjust if we discover issues. Let's just get @shantha-14 to ACK the approach.

Since this no longer adds CI-all-test for west.yml changes, we risk under-testing PRs until this list is checked by the right people. If that's worth the risk, we can announce it after merging, and tell people to open PRs to this file.

>The remaining blocker is getting the action changes merged upstream and updating this PR not to point at your fork of the action repository, right?

If we are merging with the current list, yep.

**Comment by mbolivar-nordic** [comment]
@carlescufi did you mean to merge this? The action changes are still not merged: https://github.com/zephyrproject-rtos/action-manifest/pull/2

**Comment by carlescufi** [comment]
> @carlescufi did you mean to merge this? The action changes are still not merged: [zephyrproject-rtos/action-manifest#2](https://github.com/zephyrproject-rtos/action-manifest/pull/2)

Darn, nope, my mistake. I have merged the changes in action-manifest now.

**Pull request merged**
